src/post-primary-brand/index.js

```diff
@@ -0,0 +1,117 @@
+/* global newspackPostPrimaryBrandVars */
+
+import { __, _n, _x } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import { Button, Flex, FlexItem, SelectControl } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+
+import './index.scss';
+
+/**
+ * Module Constants
+ */
+const DEFAULT_QUERY = {
+	per_page: -1,
+	orderby: 'name',
+	order: 'asc',
+	_fields: 'id,name,parent',
+	context: 'view',
+};
+
+const EMPTY_ARRAY = [];
+
+const ADMIN_URL = newspackPostPrimaryBrandVars.adminURL;
+
+const TAXONOMY_SLUG = newspackPostPrimaryBrandVars.taxonomySlug;
+
+const META_KEY = newspackPostPrimaryBrandVars.metaKey;
+
+/**
+ * Adds a primary brand selector to the post editor.
+ */
+const NewspackPostPrimaryBrand = ( { slug } ) => {
+	const { editPost } = useDispatch( 'core/editor' );
+
+	const [ currentPrimaryBrand, setCurrentPrimaryBrand ] = useState( null );
+
+	const { terms, availableTerms, taxonomy } = useSelect(
+		select => {
+			const { getCurrentPost, getEditedPostAttribute } = select( 'core/editor' );
+			const { getTaxonomy, getEntityRecords } = select( coreStore );
+			const _taxonomy = getTaxonomy( slug );
+			const post = getCurrentPost();
+
+			setCurrentPrimaryBrand( post.meta[ META_KEY ] );
```

Comment:
There's a simpler approach for this that doesn't require `useState`. The edited post is already storing that information that we can use through `useSelect()` and `editPost()`:

```js
	const { editPost } = useDispatch( 'core/editor' );

	const { terms, availableTerms, primaryBrand } = useSelect(
		select => {
			const { getEditedPostAttribute } = select( 'core/editor' );
			const { getTaxonomy, getEntityRecords } = select( coreStore );
			const _taxonomy = getTaxonomy( slug );
			const meta = getEditedPostAttribute( 'meta' );

			return {
				terms: _taxonomy ? getEditedPostAttribute( _taxonomy.rest_base ) : EMPTY_ARRAY,
				availableTerms: getEntityRecords( 'taxonomy', slug, DEFAULT_QUERY ) || EMPTY_ARRAY,
				taxonomy: _taxonomy,
				primaryBrand: meta[ META_KEY ],
			};
		},
		[ slug ]
	);
```

You'll be using the `primaryBrand` const and only required to update the edited post attribute through `editPost( { meta: { [ META_KEY ]: termId } } );`

Reply:
gotcha! done!